pipelines.go

```diff
@@ -55,6 +56,60 @@ func (p *Pipeline) Lock() *Pipeline {
 	return p
 }
 
+func (p *Pipeline) ValidateRefIds() error {
+
+	var refidsSet = make(map[string]bool)
+	var requisitestagerefSet = make(map[string]bool)
```

Comment:
Why make this a map instead of a slice?

Reply:
This simulates a set that will be used for the unique validation

Reply:
> The idiomatic way to implement a set in Go is to use a map.

![tf](https://media.giphy.com/media/op4n9nZWhQiZ2/giphy.gif)